packages/flutter/test/material/text_form_field_test.dart

```diff
@@ -816,6 +816,31 @@ void main() {
     expect(find.text('initialValue'), findsOneWidget);
   });
 
+  testWidgetsWithLeakTracking('reset resets the text fields value to the controller initial value', (WidgetTester tester) async {
+    final TextEditingController controller = TextEditingController(text: 'initialValue');
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Center(
+            child: TextFormField(
+              controller: controller,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.enterText(find.byType(TextFormField), 'changedValue');
+
+    final FormFieldState<String> state = tester.state<FormFieldState<String>>(find.byType(TextFormField));
+    state.reset();
+
+    expect(find.text('changedValue'), findsNothing);
+    expect(find.text('initialValue'), findsOneWidget);
+
+    controller.dispose();
```

Comment:
nit: you can use `addTearDown(controller.dispose)`.

Reply:
Great advise!